tests/sentry/api/test_event_search.py

```diff
@@ -768,271 +768,801 @@ def test_empty_string(self):
+            ),
+            (
+                "count():>1 AND count():<=3 OR count():>5",
+                ["or", [["and", [_c(">", 1), _c("<=", 3)]], _c(">", 5)]],
+            ),
+            (
+                "(count():>1 OR count():>2) AND count():<=3",
+                ["and", [["or", [_c(">", 1), _c(">", 2)]], _c("<=", 3)]],
+            ),
+            (
+                "(count():>1 AND count():>5) OR count():<=3",
+                ["or", [["and", [_c(">", 1), _c(">", 5)]], _c("<=", 3)]],
+            ),
+        ]
+
+        for test in tests:
+            result = get_filter(test[0])
+            assert [[test[1], "=", 1]] == result.having, test[0]
+
+    def test_aggregate_filter_and_normal_filter_in_condition(self):
+        def _eq(xy):
+            return ["equals", [["ifNull", [xy[0], "''"]], xy[1]]]
+
+        def _c(op, val):
+            return [OPERATOR_TO_FUNCTION[op], ["count", val]]
+
+        tests = [
+            ("count():>1 AND a:b", ["and", [_eq("ab")]], ["and", [_c(">", 1)]]),
+            (
+                "count():>1 AND a:b c:d",
+                ["and", [["and", [_eq("ab"), _eq("cd")]]]],
+                ["and", [_c(">", 1)]],
+            ),
+            (
+                "(a:b OR c:d) count():>1",
+                ["and", [["or", [_eq("ab"), _eq("cd")]]]],
+                ["and", [_c(">", 1)]],
+            ),
+            (
+                "(count():<3 OR count():>10) a:b c:d",
+                ["and", [["and", [_eq("ab"), _eq("cd")]]]],
+                ["and", [["or", [_c("<", 3), _c(">", 10)]]]],
+            ),
+        ]
+
+        for test in tests:
+            result = get_filter(test[0])
+            assert [[test[1], "=", 1]] == result.conditions, "cond: " + test[0]
+            assert [[test[2], "=", 1]] == result.having, "having: " + test[0]
+
+    def test_aggregate_filter_and_normal_filter_in_condition_with_or(self):
         with pytest.raises(InvalidSearchQuery) as error:
-            parse_search_query("undefined is not an object (evaluating 'function.name')") == [
-                SearchFilter(
-                    key=SearchKey(name="message"),
-                    operator="=",
-                    value=SearchValue(
-                        raw_value='undefined is not an object (evaluating "function.name")'
-                    ),
-                )
-            ]
+            get_filter("count():>1 OR a:b")
         assert (
             six.text_type(error.value)
-            == "Parse error at 'ect (evalu' (column 28). This is commonly caused by unmatched parentheses. Enclose any text in double quotes."
+            == "Having an OR between aggregate filters and normal filters is invalid."
         )
+        with pytest.raises(InvalidSearchQuery) as error:
+            get_filter("(count():>1 AND a:b) OR a:b")
+        assert (
+            six.text_type(error.value)
+            == "Having an OR between aggregate filters and normal filters is invalid."
```

Comment:
If my understanding is correct it's the `(count():>1 AND a:b)` that isn't allowed.

```suggestion
            == "Having an AND between aggregate filters and normal filters is invalid."
```

Reply:
No it's the OR that is not allowed. Aggregate filters (`count():>1`) are put in the HAVING clause of the Clickhouse query while other filters are put in the WHERE clause. Those are implicitly ANDed together, but you can't OR between the two. Thus doing an OR between an aggregate filter and normal filter isn't allowed.